ext/signals.c

```diff
@@ -102,6 +107,138 @@ static bool ddtrace_crashtracker_check_result(ddog_VoidResult result, const char
     return true;
 }
 
+#if PHP_VERSION_ID >= 80000
+static zend_never_inline ZEND_COLD void ddtrace_crasht_failed_tag_push(
+    ddog_Error *err,
+    ddog_CharSlice key
+) {
+    ddog_CharSlice msg = ddog_Error_message(err);
+    LOG(DEBUG,
+        "Failed to push tag \"%.*s\": %.*s",
+        (int) key.len, key.ptr,
+        (int) msg.len, msg.ptr);
+    ddog_Error_drop(err);
+}
+
+ // This doesn't increase the refcount of the returned string (no cleanup).
+static zend_always_inline zend_string *ddtrace_crasht_ini_get_value(ddog_CharSlice ini) {
+    zend_string *ini_zstr = zend_string_init(ini.ptr, ini.len, true);
+    zend_string *value = zend_ini_get_value(ini_zstr);
+    zend_string_release(ini_zstr);
+    return value;
+}
+
+// Pass in a key like "php.opcache.enable" and "php." will get stripped off,
+// and that's what the INI setting will be.
+static void ddtrace_crasht_add_opcache_tag(
+    ddog_Vec_Tag *tags,
+    ddog_CharSlice key
+) {
+    const ddog_CharSlice PREFIX = DDOG_CHARSLICE_C("php.");
```

Comment:
```suggestion
    const char *PREFIX = "php.";
```
Let's keep it simple and use strlen() for the length. We will just rely on the compiler to recognize that strlen(PREFIX) is 4.